core/parameters/src/config.rs

```diff
@@ -94,3 +98,111 @@ impl Default for AccountCreationConfig {
+    ///
+    /// The actual gas forwarding allowance is a linear interpolation between
+    /// [`MIN_OUTGOING_GAS`] and [`MAX_OUTGOING_GAS`], or 0 if the receiver is
+    /// fully congested.
+    pub min_outgoing_gas: Gas,
+
+    /// How much gas the chosen allowed shard can send to a 100% congested shard.
+    ///
+    /// This amount is the absolute minimum of new workload a congested shard has to
+    /// accept every round. It ensures deadlocks are provably impossible. But in
+    /// ideal conditions, the gradual reduction of new workload entering the system
+    /// combined with gradually limited forwarding to congested shards should
+    /// prevent shards from becoming 100% congested in the first place.
+    pub red_gas: Gas,
+
+    /// The maximum amount of gas in a chunk spent on converting new transactions to
+    /// receipts.
+    ///
+    /// The actual gas forwarding allowance is a linear interpolation between
+    /// [`MIN_TX_GAS`] and [`MAX_TX_GAS`], based on the incoming congestion of the
+    /// local shard. Additionally, transactions can be rejected if the receiving
+    /// remote shard is congested more than [`REJECT_TX_CONGESTION_THRESHOLD`] based
+    /// on their general congestion level.
+    pub max_tx_gas: Gas,
+
+    /// The minimum amount of gas in a chunk spent on converting new transactions
+    /// to receipts, as long as the receiving shard is not congested.
+    ///
+    /// The actual gas forwarding allowance is a linear interpolation between
+    /// [`MIN_TX_GAS`] and [`MAX_TX_GAS`], based on the incoming congestion of the
+    /// local shard. Additionally, transactions can be rejected if the receiving
+    /// remote shard is congested more than [`REJECT_TX_CONGESTION_THRESHOLD`] based
+    /// on their general congestion level.
+    pub min_tx_gas: Gas,
+
+    /// How much congestion a shard can tolerate before it stops all shards from
+    /// accepting new transactions with the receiver set to the congested shard.
+    pub reject_tx_congestion_threshold: f64,
+}
+
+// The following default constants have been defined in
+// [NEP-539](https://github.com/near/NEPs/pull/539) after extensive fine-tuning
+// and discussions.
+
+impl Default for CongestionControlConfig {
+    fn default() -> Self {
+        const PGAS: Gas = 10u64.pow(15);
+        const TGAS: Gas = 10u64.pow(12);
+
+        Self {
+            max_congestion_incoming_gas: 20 * PGAS,
+            max_congestion_outgoing_gas: 2 * PGAS,
+            max_congestion_memory_consumption: bytesize::ByteSize::mb(1000u64).0,
+            max_congestion_missed_chunks: 10,
+            max_outgoing_gas: 300 * PGAS,
+            min_outgoing_gas: 1 * PGAS,
+            red_gas: 1 * PGAS,
+            max_tx_gas: 500 * TGAS,
+            min_tx_gas: 20 * TGAS,
+            reject_tx_congestion_threshold: 0.25,
+        }
+    }
+}
+
+impl Eq for CongestionControlConfig {
+    // TODO(congestion_control) - implement Eq for CongestionControlConfig
```

Comment:
This should be resolved before merging.

Reply:
I was planning to do something here but now I actually think this is the best approach. 

The Eq trait doesn't have any methods to be implemented. It merely adds more "meaning" to the methods already implemented by the PartialEq trait. 

Normally Eq is not defined for f64 because NaN != NaN. Here we don't expect any NaN's. My suggestion would be to keep the Eq implementation, add an assertion that the relevant config is not NaN and add a nice comment explaining what's up to the Eq implementation for the Config struct. 

@jakmeier What do you think? 

I also considered adding a new type alias `type f64_eq = f64` or `struct f64_eq(f64)` but they both boil down to the same - needing to add Eq on some float somewhere.

Reply:
Agreed, it's fine to use this implementation with explanation in comments. Personally, I wouldn't wrap it in `struct f64_eq(f64)` but it's a style question.

An alternative would be to use `Rational32`  instead of `f64`, like we do in other runtime and genesis config. Again, mostly a style question what you prefer personally. Just wanted to mention that this exists.

Reply:
Thanks, I stumbled across Rational32 when working on the params and it may be a good alternative. I'll have a look if it's any better if I have some spare time.